libs/payments/src/lib/payment-provider/bexio/bexioPaymentProvider.ts

```diff
@@ -0,0 +1,368 @@
+    this.invoiceTitleRenewalMembership = props.invoiceTitleRenewalMembership
+    this.invoiceMailSubjectNewMembership = props.invoiceMailSubjectNewMembership
+    this.invoiceMailBodyNewMembership = props.invoiceMailBodyNewMembership
+    this.invoiceMailSubjectRenewalMembership = props.invoiceMailSubjectRenewalMembership
+    this.invoiceMailBodyRenewalMembership = props.invoiceMailBodyRenewalMembership
+    this.markInvoiceAsOpen = props.markInvoiceAsOpen
+    this.prisma = props.prisma
+    this.bexio = new Bexio(this.apiKey)
+  }
+
+  async webhookForPaymentIntent(props: WebhookForPaymentIntentProps): Promise<IntentState[]> {
+    throw new WebhookNotImplementedError()
+  }
+
+  /**
+   * Creates a payment intent based on the provided invoice details.
+   * If the invoice has an associated subscription ID, it throws an error.
+   * Otherwise, it proceeds to create a payment intent using Bexio's API.
+   *
+   * @param {CreatePaymentIntentProps} props - The properties required to create the payment intent.
+   * @returns {Promise<Intent>} A promise that resolves to the created payment intent object.
+   * @throws {NoSubscriptionIdInInvoice} Throws an error if the invoice contains a subscription ID.
+   */
+  async createIntent(props: CreatePaymentIntentProps): Promise<Intent> {
+    if (props.invoice.subscriptionID) {
+      throw new NoSubscriptionIdInInvoice()
+    }
+    const created = await this.bexioCreate(props.invoice.id, false)
+    return created
+  }
+
+  /**
+   * This function calls the `bexioCreate` method to create an invoice in Bexio.
+   * It is designed to handle the creation of remote invoices, specifically for renewals.
+   *
+   * @async
+   * @function
+   * @param {CreateRemoteInvoiceProps} props - The properties required to create the remote invoice.
+   * @param {string} props.invoice.id - The ID of the invoice.
+   *
+   * @returns {Promise<void>} Resolves when the remote invoice is successfully created.
+   */
+  async createRemoteInvoice(props: CreateRemoteInvoiceProps) {
+    await this.bexioCreate(props.invoice.id, true)
+  }
+
+  /**
+   * Checks the status of a payment intent in Bexio.
+   * It makes a direct fetch request to the Bexio API because the Bexio library
+   * does not return the status when querying for an invoice.
+   *
+   * @param {CheckIntentProps} CheckIntentProps - An object with the intentID to check the status.
+   * @returns {Promise<IntentState>} A promise that resolves to the status of the intent.
+   * @throws {ResponseNOK} Throws an error if the response status from Bexio is not 200.
+   * @throws {UnknownIntentState} Throws an error if the intent status is not recognized.
+   */
+  async checkIntentStatus({intentID}: CheckIntentProps): Promise<IntentState> {
+    // currently the bexio library we use doesn't return the status (kb_item_status_id)
+    // when querying for invoice, so we need to do it "manually" using fetch api
```

Comment:
Maybe discuss adding this to the bexio npm library and if not, then at least put the code in a separate function to keep the code dry for the future.